### PR TITLE
Merge 1.4 into master

### PR DESCRIPTION
**Pull request by ClementGautier**
| Q                 | A
| ----------------- | ---
| Added Specs       | N/A
| Added Behats      | N/A
| Travis CI is ok   | Running
| Changelog updated | Y

### CODE REVIEW

File: features/Context/Page/UserRole/Edit.php [hunk 1]
```diff
@@ -15,4 +16,55 @@ class Edit extends Form
...
+    /**
+     * Select the specified $group
+     *
+     * @param string $group
+     */
+    public function selectGroup($group)
+    {
+        $element = $this->spin(function () use ($group) {
+            return $this->getElement('ACL Group')->find('css', sprintf('a:contains("%s")', $group));
+        }, 5, sprintf('Unable to find the ACL group', $group));
```
Comment (sumbobyboys) [inline features/Context/Page/UserRole/Edit.php]: this is not compatible with master

File: features/Context/Page/UserRole/Edit.php [hunk 2]
```diff
@@ -15,4 +16,55 @@ class Edit extends Form
...
+     *
+     * @param string $role
+     *
+     * @throws ElementNotFoundException
+     */
+    public function selectRole($role)
+    {
+        $node = $this->spin(function () use ($role) {
+            return $this->getElement('ACL Role')->find('css', sprintf('strong:contains("%s")', $role));
+        }, 5, sprintf('Unable to find the ACL role', $role));
```
Comment (sumbobyboys) [inline features/Context/Page/UserRole/Edit.php]: same here

Comment (sumbobyboys) [inline features/Context/Page/UserRole/Edit.php]: and this sprintf does not work

File: features/Context/Page/UserRole/Edit.php (hunk 1 above)
Comment (sumbobyboys) [inline features/Context/Page/UserRole/Edit.php]: this sprintf does not make any sens :)

**Comment by jjanvier** [comment]
gtm

File: features/Context/Page/UserRole/Edit.php (hunk 1 above)
Comment (jjanvier) [inline features/Context/Page/UserRole/Edit.php]: we can fix on 1.4, don't block this pr for that imo

File: features/Context/Page/UserRole/Edit.php (hunk 2 above)
Comment (jjanvier) [inline features/Context/Page/UserRole/Edit.php]: we can fix on 1.4, don't block this pr for that imo

File: src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php [hunk 3]
```diff
@@ -32,36 +29,19 @@ class AddVersionSubscriber implements EventSubscriber
...
-    /** @var ChainedUpdateGuesser */
-    protected $guesser;
-
-    /** @var NormalizerInterface */
-    protected $normalizer;
-
-    /** @var VersionContext */
-    protected $versionContext;
+    /** @var ContainerInterface */
+    protected $container;
```
Comment (sumbobyboys) [inline src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php]: could we fix that ?

**Comment by ClementGautier** [comment]
@sumbobyboys Agree that the sprintf is not well formed but it will not output any error. It will be fixed by @jjanvier soon directly on 1.4 to avoid merge conflict during the next merge 1.4 into master.

File: src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php (hunk 3 above)
Comment (ClementGautier) [inline src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php]: nop, no easy workaround for this thing actually. As we discussed the good way would be to not listen on doctrine events but on our events, that would fix the circular reference error but its a heavy work ;)

**Comment by skeleton** [comment]
GTM

**Pull request merged**